<a href="https://colab.research.google.com/github/overgroove/samsung_lecture/blob/main/01_feature_eng_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Introduce of dataset**

- **Ames Housing Dataset**은 미국 아이오와주 에임스 지역의 주택 판매 데이터입니다.  
- 총 79개의 설명 변수(Feature)와 1개의 타겟 변수(Target)로 구성
- 데이터의 유형이 매우 다양하고 현실적인 전처리가 필요
- 머신러닝 필드에서 regression을 위한 표준 데이터셋으로 많이 활용
- https://www.openml.org/search?type=data&status=active&id=42165

## **data loading**

In [1]:
# 필요모듈 import
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.datasets import fetch_openml

pd.options.display.max_columns = 200
pd.options.display.max_rows = 200

In [5]:
# fetch_openml로 'house_prices' 데이터셋 로딩
data = fetch_openml('house_prices', as_frame=True)
df = pd.DataFrame(data['data'])
df.head()

,Id,MSSubClass,MSZoning,LotFrontage,LotArea,Street,Alley,LotShape,LandContour,Utilities,LotConfig,LandSlope,Neighborhood,Condition1,Condition2,BldgType,HouseStyle,OverallQual,OverallCond,YearBuilt,YearRemodAdd,RoofStyle,RoofMatl,Exterior1st,Exterior2nd,MasVnrType,MasVnrArea,ExterQual,ExterCond,Foundation,BsmtQual,BsmtCond,BsmtExposure,BsmtFinType1,BsmtFinSF1,BsmtFinType2,BsmtFinSF2,BsmtUnfSF,TotalBsmtSF,Heating,HeatingQC,CentralAir,Electrical,1stFlrSF,2ndFlrSF,LowQualFinSF,GrLivArea,BsmtFullBath,BsmtHalfBath,FullBath,HalfBath,BedroomAbvGr,KitchenAbvGr,KitchenQual,TotRmsAbvGrd,Functional,Fireplaces,FireplaceQu,GarageType,GarageYrBlt,GarageFinish,GarageCars,GarageArea,GarageQual,GarageCond,PavedDrive,WoodDeckSF,OpenPorchSF,EnclosedPorch,3SsnPorch,ScreenPorch,PoolArea,PoolQC,Fence,MiscFeature,MiscVal,MoSold,YrSold,SaleType,SaleCondition
0,1,60,RL,65.0,8450,Pave,NaN,Reg,Lvl,AllPub,Inside,Gtl,CollgCr,Norm,Norm,1Fam,2Story,7,5,2003,2003,Gable,CompShg,VinylSd,VinylSd,BrkFace,196.0,Gd,TA,PConc,Gd,TA,No,GLQ,706,Unf,0,150,856,GasA,Ex,Y,SBrkr,856,854,0,1710,1,0,2,1,3,1,Gd,8,Typ,0,NaN,Attchd,2003.0,RFn,2,548,TA,TA,Y,0,61,0,0,0,0,NaN,NaN,NaN,0,2,2008,WD,Normal
1,2,20,RL,80.0,9600,Pave,NaN,Reg,Lvl,AllPub,FR2,Gtl,Veenker,Feedr,Norm,1Fam,1Story,6,8,1976,1976,Gable,CompShg,MetalSd,MetalSd,None,0.0,TA,TA,CBlock,Gd,TA,Gd,ALQ,978,Unf,0,284,1262,GasA,Ex,Y,SBrkr,1262,0,0,1262,0,1,2,0,3,1,TA,6,Typ,1,TA,Attchd,1976.0,RFn,2,460,TA,TA,Y,298,0,0,0,0,0,NaN,NaN,NaN,0,5,2007,WD,Normal
2,3,60,RL,68.0,11250,Pave,NaN,IR1,Lvl,AllPub,Inside,Gtl,CollgCr,Norm,Norm,1Fam,2Story,7,5,2001,2002,Gable,CompShg,VinylSd,VinylSd,BrkFace,162.0,Gd,TA,PConc,Gd,TA,Mn,GLQ,486,Unf,0,434,920,GasA,Ex,Y,SBrkr,920,866,0,1786,1,0,2,1,3,1,Gd,6,Typ,1,TA,Attchd,2001.0,RFn,2,608,TA,TA,Y,0,42,0,0,0,0,NaN,NaN,NaN,0,9,2008,WD,Normal
3,4,70,RL,60.0,9550,Pave,NaN,IR1,Lvl,AllPub,Corner,Gtl,Crawfor,Norm,Norm,1Fam,2Story,7,5,1915,1970,Gable,CompShg,'Wd Sdng','Wd Shng',None,0.0,TA,TA,BrkTil,TA,Gd,No,ALQ,216,Unf,0,540,756,GasA,Gd,Y,SBrkr,961,756,0,1717,1,0,1,0,3,1,Gd,7,Typ,1,Gd,Detchd,1998.0,Unf,3,642,TA,TA,Y,0,35,272,0,0,0,NaN,NaN,NaN,0,2,2006,WD,Abnorml
4,5,60,RL,84.0,14260,Pave,NaN,IR1,Lvl,AllPub,FR2,Gtl,NoRidge,Norm,Norm,1Fam,2Story,8,5,2000,2000,Gable,CompShg,VinylSd,VinylSd,BrkFace,350.0,Gd,TA,PConc,Gd,TA,Av,GLQ,655,Unf,0,490,1145,GasA,Ex,Y,SBrkr,1145,1053,0,2198,1,0,2,1,4,1,Gd,9,Typ,1,TA,Attchd,2000.0,RFn,3,836,TA,TA,Y,192,84,0,0,0,0,NaN,NaN,NaN,0,12,2008,WD,Normal


In [3]:
# 로딩 데이터 데이터프레임으로 할당


In [6]:
# 데이터 설명 출력
print(data['DESCR'])

Ask a home buyer to describe their dream house, and they probably won't begin with the height of the basement ceiling or the proximity to an east-west railroad. But this playground competition's dataset proves that much more influences price negotiations than the number of bedrooms or a white-picket fence.

With 79 explanatory variables describing (almost) every aspect of residential homes in Ames, Iowa, this competition challenges you to predict the final price of each home.

MSSubClass: Identifies the type of dwelling involved in the sale.	

        20	1-STORY 1946 & NEWER ALL STYLES
        30	1-STORY 1945 & OLDER
        40	1-STORY W/FINISHED ATTIC ALL AGES
        45	1-1/2 STORY - UNFINISHED ALL AGES
        50	1-1/2 STORY FINISHED ALL AGES
        60	2-STORY 1946 & NEWER
        70	2-STORY 1945 & OLDER
        75	2-1/2 STORY ALL AGES
        80	SPLIT OR MULTI-LEVEL
        85	SPLIT FOYER
        90	DUPLEX - ALL STYLES AND AGES
       120	1-STORY PUD (Planned Unit Development) - 1

## Ames Housing Datase 변수 상세 설명서 (79개 설명 변수)

총 79개의 변수와 1개의 타겟 변수(`SalePrice`)로 구성되어 있습니다.

---

### 1. 기본 정보 및 입지 (General & Location)
| 변수 이름 | 변수 명(한글) | 변수의 의미 및 특징 |
| :--- | :--- | :--- |
| **MSSubClass** | 주거 유형 | 판매와 관련된 주거 유형 (예: 1층 건물, 2층 건물 등) |
| **MSZoning** | 판매 구역 분류 | 일반적인 구역 분류 (공공시설, 주거지역, 상업지역 등) |
| **LotFrontage** | 도로 인접 거리 | 부지에 연결된 도로의 직선 거리 (단위: 피트) |
| **LotArea** | 부지 면적 | 평방 피트 단위의 부지 크기 |
| **Street** | 도로 접근 유형 | 도로 접근 유형 (Pave-포장, Grvl-자갈) |
| **Alley** | 골목 접근 유형 | 부지에 접근하는 골목의 유형 |
| **LotShape** | 부지 모양 | 부지의 일반적인 모양 (정형, 부정형 등) |
| **LandContour** | 지형 평탄도 | 부지의 평탄함 정도 (평지, 경사지 등) |
| **Utilities** | 공공시설 유형 | 이용 가능한 공공시설 (전기, 가스, 수도 등) |
| **LotConfig** | 부지 구성 | 부지의 위치적 구성 (Corner, Inside 등) |
| **LandSlope** | 부지 경사도 | 부지의 경사 정도 (완만함, 가파름 등) |
| **Neighborhood** | 동네 이름 | 에임스 시 내의 물리적 위치 |
| **Condition1** | 인접 조건 1 | 주요 도로 또는 철도와의 인접성 정보 |
| **Condition2** | 인접 조건 2 | 여러 조건이 있을 경우의 두 번째 인접성 정보 |

### 2. 주택 유형 및 품질 (Type & Quality)
| 변수 이름 | 변수 명(한글) | 변수의 의미 및 특징 |
| :--- | :--- | :--- |
| **BldgType** | 주거 형태 | 주거지 형태 (단독주택, 타운하우스 등) |
| **HouseStyle** | 주택 스타일 | 주거 스타일 (1층형, 2층형 등) |
| **OverallQual** | 전반적 품질 | 주택의 전체적인 마감 및 재질 품질 (1~10점) |
| **OverallCond** | 전반적 상태 | 주택의 전반적인 상태 등급 (1~10점) |
| **YearBuilt** | 완공 연도 | 주택의 원래 완공일 |
| **YearRemodAdd** | 리모델링 연도 | 리모델링 날짜 (리모델링이 없으면 완공일과 동일) |
| **RoofStyle** | 지붕 형태 | 지붕의 유형 (Gable, Hip 등) |
| **RoofMatl** | 지붕 재료 | 지붕을 덮은 재료 |
| **Exterior1st** | 외부 마감 1 | 주택 외부 덮개 재료 1 |
| **Exterior2nd** | 외부 마감 2 | 주택 외부 덮개 재료 2 (두 개 이상일 경우) |
| **MasVnrType** | 석재 외장 유형 | 돌이나 벽돌 외장의 유형 |
| **MasVnrArea** | 석재 외장 면적 | 평방 피트 단위의 석재 외장 면적 |
| **ExterQual** | 외장 품질 | 외부 재료의 품질 |
| **ExterCond** | 외장 상태 | 외부 재료의 현재 상태 |
| **Foundation** | 기초 공사 유형 | 기초 공사에 사용된 재료 및 유형 |

### 3. 지하실 정보 (Basement)
| 변수 이름 | 변수 명(한글) | 변수의 의미 및 특징 |
| :--- | :--- | :--- |
| **BsmtQual** | 지하실 높이 품질 | 지하실의 높이와 관련된 품질 등급 |
| **BsmtCond** | 지하실 상태 | 지하실의 전반적인 상태 |
| **BsmtExposure** | 지하실 노출도 | 정원이나 실외로 연결되는 벽의 노출 정도 |
| **BsmtFinType1** | 지하실 마감 상태 1 | 지하실 완성 면적의 등급 1 |
| **BsmtFinSF1** | 지하실 마감 면적 1 | 평방 피트 단위의 마감 면적 1 |
| **BsmtFinType2** | 지하실 마감 상태 2 | 두 번째 마감 면적의 등급 (있는 경우) |
| **BsmtFinSF2** | 지하실 마감 면적 2 | 평방 피트 단위의 마감 면적 2 |
| **BsmtUnfSF** | 미완성 지하실 면적 | 마감되지 않은 지하실 평방 피트 |
| **TotalBsmtSF** | 총 지하실 면적 | 지하실의 전체 평방 피트 면적 |

### 4. 실내 설비 및 면적 (Interior & Area)
| 변수 이름 | 변수 명(한글) | 변수의 의미 및 특징 |
| :--- | :--- | :--- |
| **Heating** | 난방 유형 | 난방 시스템의 종류 |
| **HeatingQC** | 난방 품질 및 상태 | 난방의 품질과 현재 상태 |
| **CentralAir** | 중앙 냉방 여부 | 중앙 냉방 시스템 유무 (Y/N) |
| **Electrical** | 전기 시스템 | 전기 시스템의 종류 |
| **1stFlrSF** | 1층 면적 | 1층 평방 피트 면적 |
| **2ndFlrSF** | 2층 면적 | 2층 평방 피트 면적 |
| **LowQualFinSF** | 저품질 마감 면적 | 모든 층에서 마감 품질이 낮은 면적 |
| **GrLivArea** | 지상 거주 면적 | 지상 전체 거주 평방 피트 (매우 중요) |
| **BsmtFullBath** | 지하실 풀-화장실 | 지하실에 있는 전체 화장실 개수 |
| **BsmtHalfBath** | 지하실 하프-화장실 | 지하실에 있는 세면대 위주의 화장실 개수 |
| **FullBath** | 지상 풀-화장실 | 지상에 있는 전체 화장실 개수 |
| **HalfBath** | 지상 하프-화장실 | 지상에 있는 세면대 위주의 화장실 개수 |
| **BedroomAbvGr** | 지상 침실 수 | 지상에 있는 침실 개수 |
| **KitchenAbvGr** | 지상 주방 수 | 지상에 있는 주방 개수 |
| **KitchenQual** | 주방 품질 | 주방 마감 및 설비의 품질 |
| **TotRmsAbvGrd** | 지상 총 방 개수 | 화장실을 제외한 지상 전체 방의 개수 |
| **Functional** | 주택 기능성 | 주택의 기능적 손상 여부 및 등급 |
| **Fireplaces** | 벽난로 개수 | 설치된 벽난로의 개수 |
| **FireplaceQu** | 벽난로 품질 | 벽난로의 상태 및 품질 |

### 5. 차고 및 부속 시설 (Garage & Misc)
| 변수 이름 | 변수 명(한글) | 변수의 의미 및 특징 |
| :--- | :--- | :--- |
| **GarageType** | 차고 위치/유형 | 주택과의 연결성 등 차고의 종류 |
| **GarageYrBlt** | 차고 완공 연도 | 차고가 지어진 연도 |
| **GarageFinish** | 차고 내부 마감 | 차고 내부의 인테리어 마감 상태 |
| **GarageCars** | 차고 수용 대수 | 차고에 주차 가능한 차량 대수 |
| **GarageArea** | 차고 면적 | 평방 피트 단위의 차고 크기 |
| **GarageQual** | 차고 품질 | 차고의 건축 품질 |
| **GarageCond** | 차고 상태 | 차고의 현재 관리 상태 |
| **PavedDrive** | 진입로 포장 여부 | 포장된 진입로 유무 |
| **WoodDeckSF** | 나무 데크 면적 | 나무 데크의 평방 피트 면적 |
| **OpenPorchSF** | 개방형 포치 면적 | 개방된 포치의 평방 피트 면적 |
| **EnclosedPorch** | 폐쇄형 포치 면적 | 사방이 막힌 포치의 평방 피트 면적 |
| **3SsnPorch** | 3계절 포치 면적 | 3계절용 포치의 평방 피트 면적 |
| **ScreenPorch** | 스크린 포치 면적 | 스크린(방충망 등)이 있는 포치 면적 |
| **PoolArea** | 수영장 면적 | 수영장 평방 피트 면적 |
| **PoolQC** | 수영장 품질 | 수영장의 관리 및 설비 품질 |
| **Fence** | 울타리 품질 | 울타리의 종류 및 상태 |
| **MiscFeature** | 기타 특징 | 다른 범주에 포함되지 않은 기타 특징 |
| **MiscVal** | 기타 특징 가치 | 기타 특징의 달러 가치 |

### 6. 판매 정보 (Sales Information)
| 변수 이름 | 변수 명(한글) | 변수의 의미 및 특징 |
| :--- | :--- | :--- |
| **MoSold** | 판매 월 | 주택이 판매된 월 |
| **YrSold** | 판매 연도 | 주택이 판매된 연도 |
| **SaleType** | 판매 유형 | 판매 방식 (신축, 일반 매매 등) |
| **SaleCondition** | 판매 조건 | 판매 조건 (정상 거래, 경매 등) |
| **SalePrice** | 판매 가격 | **(Target)** 최종 주택 판매 가격 (단위: 달러) |

## Ames Housing Dataset 범주형 변수 통합 마스터 사전

이 표는 데이터 설명서(DESCR)를 기반으로 **변수명, 변수의 의미, 고유값의 상세 내용**을 한국어로 정리한 것입니다. 피처 엔지니어링 실습 시 데이터 전처리 전략을 세우는 가이드로 활용하세요.

---

### 1. 토지 및 입지 관련 변수
| 변수 이름 | 변수 의미 (Korean) | 고유값 및 상세 의미 (Values & Meanings) |
| :--- | :--- | :--- |
| **MSSubClass** | 주거 유형 코드 | **20**: 1층 신축, **30**: 1층 구옥, **40**: 1층(다락 완성), **45**: 1.5층(미완성), **50**: 1.5층(완성), **60**: 2층 신축, **70**: 2층 구옥, **75**: 2.5층, **80**: 분할 레벨, **85**: 스플릿 포이어, **90**: 2가구 주택, **120**: 1층 PUD, **150**: 1.5층 PUD, **160**: 2층 PUD, **180**: 다층 PUD, **190**: 2가구 개조 주택 |
| **MSZoning** | 용도 지역 분류 | **A**: 농업, **C**: 상업, **FV**: 부유 마을, **I**: 공업, **RH**: 고밀도 주거, **RL**: 저밀도 주거, **RP**: 저밀도 주거 공원, **RM**: 중밀도 주거 |
| **Street** | 도로 접근 유형 | **Grvl**: 자갈길, **Pave**: 포장도로 |
| **Alley** | 골목 접근 유형 | **Grvl**: 자갈길, **Pave**: 포장도로, **NA**: 골목 접근 불가능 |
| **LotShape** | 부지 모양 | **Reg**: 정형, **IR1**: 약간 부정형, **IR2**: 보통 부정형, **IR3**: 완전 부정형 |
| **LandContour** | 지형 평탄도 | **Lvl**: 평지, **Bnk**: 둑(도로보다 높음), **HLS**: 경사지(옆면 경사), **Low**: 저지대 |
| **Utilities** | 공공 설비 종류 | **AllPub**: 모든 설비, **NoSewr**: 하수도 제외, **NoSeWa**: 수도/하수도 제외, **ELO**: 전기만 가능 |
| **LotConfig** | 부지 구성 위치 | **Inside**: 안쪽 부지, **Corner**: 모퉁이 부지, **CulDSac**: 막다른 길, **FR2**: 2면 도로 접함, **FR3**: 3면 도로 접함 |
| **LandSlope** | 부지 경사도 | **Gtl**: 완만함, **Mod**: 보통 경사, **Sev**: 급경사 |
| **Neighborhood** | 동네 위치 | 에임스 시 내 25개 구역 이름 (CollgCr, Veenker, Crawfor, NoRidge, Mitchel 등) |
| **Condition1/2** | 인접 조건 | **Artery**: 간선도로, **Feedr**: 보조 간선도로, **Norm**: 보통, **RRNn/RRAn**: 남북 철도, **PosN/PosA**: 공원/녹지 등 긍정적 시설, **RRNe/RRAe**: 동서 철도 |

### 2. 건물 구조 및 외장 관련 변수
| 변수 이름 | 변수 의미 (Korean) | 고유값 및 상세 의미 (Values & Meanings) |
| :--- | :--- | :--- |
| **BldgType** | 주거 형태 | **1Fam**: 단독 주택, **2FmCon**: 2가구 개조, **Duplx**: 복층 주택, **TwnhsE**: 타운하우스(끝집), **TwnhsI**: 타운하우스(안쪽집) |
| **HouseStyle** | 주택 스타일 | **1Story**: 1층형, **1.5Fin/Unf**: 1.5층, **2Story**: 2층형, **2.5Fin/Unf**: 2.5층, **SFoyer**: 스플릿 포이어, **SLvl**: 스플릿 레벨 |
| **RoofStyle** | 지붕 형태 | **Flat**: 평지붕, **Gable**: 박공지붕, **Gambrel**: 박공(역육각), **Hip**: 모임지붕, **Mansard**: 맨사드 지붕, **Shed**: 외경사 지붕 |
| **RoofMatl** | 지붕 재료 | **ClyTile**: 점토/타일, **CompShg**: 아스팔트 싱글, **Membran**: 멤브레인, **Metal**: 금속, **Roll**: 롤 형태, **Tar&Grv**: 타르/자갈, **WdShake**: 나무 쉐이크, **WdShngl**: 나무 싱글 |
| **Exterior1st/2nd**| 외벽 마감재 | **AsbShng**: 석면 싱글, **BrkComm**: 일반 벽돌, **BrkFace**: 치장 벽돌, **CBlock**: 시멘트 블록, **CemntBd**: 시멘트 보드, **HdBoard**: 하드보드, **ImStucc**: 인조 스투코, **MetalSd**: 금속 사이딩, **VinylSd**: 비닐 사이딩, **Wd Sdng**: 목재 사이딩 등 |
| **MasVnrType** | 외장 석재 유형 | **BrkCmn**: 일반 벽돌, **BrkFace**: 치장 벽돌, **CBlock**: 시멘트 블록, **None**: 없음, **Stone**: 석재 |
| **Foundation** | 기초 공사 유형 | **BrkTil**: 벽돌/타일, **CBlock**: 시멘트 블록, **PConc**: 콘크리트 타설, **Slab**: 슬래브 기초, **Stone**: 석재 기초, **Wood**: 목재 기초 |

### 3. 내부 품질 및 설비 관련 변수 (순서형/등급)
| 변수 이름 | 변수 의미 (Korean) | 고유값 및 상세 의미 (Values & Meanings) |
| :--- | :--- | :--- |
| **ExterQual/Cond** | 외장 품질/상태 | **Ex**: 최고, **Gd**: 좋음, **TA**: 보통, **Fa**: 낮음, **Po**: 최악 |
| **BsmtQual** | 지하실 높이 | **Ex**: 100인치 이상, **Gd**: 90-99, **TA**: 80-89, **Fa**: 70-79, **Po**: 70미만, **NA**: 지하실 없음 |
| **BsmtCond** | 지하실 상태 | **Ex**: 최고, **Gd**: 좋음, **TA**: 보통(약간 습함), **Fa**: 낮음(균열/습기), **Po**: 최악(심한 균열/침수), **NA**: 지하실 없음 |
| **BsmtExposure** | 지하실 노출도 | **Gd**: 좋음, **Av**: 보통, **Mn**: 최소 노출, **No**: 노출 없음, **NA**: 지하실 없음 |
| **BsmtFinType1/2**| 지하실 마감 수준 | **GLQ**: 우수 거주용, **ALQ**: 평균 거주용, **BLQ**: 평균 이하, **Rec**: 레크리에이션 룸, **LwQ**: 저품질 마감, **Unf**: 미완성, **NA**: 지하실 없음 |
| **HeatingQC** | 난방 품질/상태 | **Ex**: 최고, **Gd**: 좋음, **TA**: 보통, **Fa**: 낮음, **Po**: 최악 |
| **CentralAir** | 중앙 냉방 여부 | **N**: 없음, **Y**: 있음 |
| **Electrical** | 전기 시스템 | **SBrkr**: 표준 차단기, **FuseA**: 60A 이상 퓨즈(우수), **FuseF**: 60A 퓨즈(보통), **FuseP**: 퓨즈(불량), **Mix**: 혼합형 |
| **KitchenQual** | 주방 시설 품질 | **Ex**: 최고, **Gd**: 좋음, **TA**: 보통, **Fa**: 낮음, **Po**: 최악 |
| **Functional** | 주택 기능성 | **Typ**: 정상, **Min1/2**: 미세 결함, **Mod**: 보통 결함, **Maj1/2**: 주요 결함, **Sev**: 심각한 손상, **Sal**: 폐기 수준 |
| **FireplaceQu** | 벽난로 품질 | **Ex**: 최고, **Gd**: 좋음, **TA**: 보통, **Fa**: 낮음, **Po**: 최악, **NA**: 벽난로 없음 |

### 4. 차고 및 부속 시설 관련 변수
| 변수 이름 | 변수 의미 (Korean) | 고유값 및 상세 의미 (Values & Meanings) |
| :--- | :--- | :--- |
| **GarageType** | 차고 위치 | **2Types**: 2종 이상, **Attchd**: 주택 부착형, **Basment**: 지하실 차고, **BuiltIn**: 내장형, **CarPort**: 카포트, **Detchd**: 주택 분리형, **NA**: 차고 없음 |
| **GarageFinish** | 차고 내부 마감 | **Fin**: 완성, **RFn**: 거친 마감, **Unf**: 미완성, **NA**: 차고 없음 |
| **GarageQual/Cond**| 차고 품질/상태 | **Ex**: 최고, **Gd**: 좋음, **TA**: 보통, **Fa**: 낮음, **Po**: 최악, **NA**: 차고 없음 |
| **PavedDrive** | 진입로 포장 여부 | **Y**: 포장됨, **P**: 부분 포장, **N**: 비포장 |
| **PoolQC** | 수영장 품질 | **Ex**: 최고, **Gd**: 좋음, **TA**: 보통, **Fa**: 낮음, **NA**: 수영장 없음 |
| **Fence** | 울타리 품질 | **GdPrv**: 우수한 프라이버시, **MnPrv**: 최소 프라이버시, **GdWo**: 우수한 나무 울타리, **MnWw**: 최소 나무/철사 울타리, **NA**: 울타리 없음 |
| **MiscFeature** | 기타 부대시설 | **Elev**: 엘리베이터, **Gar2**: 제2차고, **Othr**: 기타, **Shed**: 창고, **TenC**: 테니스 코트, **NA**: 없음 |

### 5. 거래 정보 관련 변수
| 변수 이름 | 변수 의미 (Korean) | 고유값 및 상세 의미 (Values & Meanings) |
| :--- | :--- | :--- |
| **SaleType** | 판매 유형 | **WD**: 보증서 일반, **CWD**: 현금 보증서, **VWD**: VA 융자 보증서, **New**: 신축, **COD**: 법원 관리인 증서, **Con**: 계약 매매 등 |
| **SaleCondition** | 판매 조건 | **Normal**: 정상, **Abnorml**: 비정상(경매/압류), **AdjLand**: 인접 토지 매입, **Alloca**: 할당 판매, **Family**: 가족 간 거래, **Partial**: 완공 전 판매 |